crates/tauri-bundler/src/bundle/macos/app.rs

```diff
@@ -214,7 +214,11 @@ fn create_info_plist(
     settings.bundle_identifier().into(),
   );
   plist.insert("CFBundleInfoDictionaryVersion".into(), "6.0".into());
-  plist.insert("CFBundleName".into(), settings.product_name().into());
+  if let Some(bundle_name) = settings.macos().bundle_name.clone() {
+    plist.insert("CFBundleName".into(), bundle_name.into());
+  } else {
+    plist.insert("CFBundleName".into(), settings.product_name().into());
+  }
```

Comment:
This code can be "simplified" (not really simpler) to how it was done for CFBundleVersion below so like this
```suggestion
  plist.insert(
    "CFBundleName".into(),
    settings
      .macos()
      .bundle_name
      .as_deref()
      .unwrap_or_else(|| settings.product_name())
      .into(),
  );
```